### PR TITLE
Merge Spack changes and update Uberenv/Spack/SCR

### PR DESCRIPTION
**Pull request by white238**
* SCR 3.0rc2 is marked deprecated
* Uberenv now has uses the `--fresh` option to work around changing the default to `--reuse`
* Bump Spack

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

**Comment by white238** [comment]
Don't merge this. As it usually does this spiraled into more things. SCR needed to be updated so versions are changing.

**Comment by white238** [comment]
this is fine to go now... though maybe re-reviewed?

File: host-configs/rzgenie-toss_3_x86_64_ib-intel@19.0.4.cmake [hunk 1]
```diff
@@ -79,13 +79,13 @@ set(MFEM_DIR "${TPL_ROOT}/mfem-4.4.0" CACHE PATH "")
 
 set(HDF5_DIR "${TPL_ROOT}/hdf5-1.8.22" CACHE PATH "")
 
-set(LUA_DIR "/usr/WS1/axom/libs/toss_3_x86_64_ib/2022_07_05_14_13_56/gcc-8.1.0/lua-5.3.5" CACHE PATH "")
+set(LUA_DIR "${TPL_ROOT}/lua-5.3.5" CACHE PATH "")
```
Comment (kennyweiss) [inline host-configs/rzgenie-toss_3_x86_64_ib-intel@19.0.4.cmake:82]: 👍

**Review by kennyweiss — APPROVED** [review]
Re-reviewed -- thanks @white238 !

**Review by rhornung67 — APPROVED** [review]
Thank you @white238

File: scripts/spack/packages/axom/package.py [hunk 2]
```diff
@@ -31,57 +30,61 @@ def get_spec_path(spec, package_name, path_replacements={}, use_bin=False):
...
+    version("0.4.0", tag="v0.4.0")
+    version("0.3.3", tag="v0.3.3")
+    version("0.3.2", tag="v0.3.2")
+    version("0.3.1", tag="v0.3.1")
+    version("0.3.0", tag="v0.3.0")
+    version("0.2.9", tag="v0.2.9")
 
-    patch('scr_examples_gtest.patch', when='@0.6.0:0.6.1')
+    # https://github.com/spack/spack/issues/31829
+    patch("examples-oneapi.patch", when="@0.6.1 +examples %oneapi")
```
Comment (bmhan12) [inline scripts/spack/packages/axom/package.py:55]: Do we need to add this patch file to axom repo?
I don't see it in the PR files.

**Review by bmhan12 — APPROVED** [review]

File: scripts/spack/packages/axom/package.py (hunk 2 above)
Comment (white238) [reply scripts/spack/packages/axom/package.py:55]: It exists in the Spack repo

**Pull request merged**